src/Core/src/Platform/Android/EditTextExtensions.cs

```diff
@@ -80,21 +79,30 @@ public static void UpdateIsTextPredictionEnabled(this AppCompatEditText editText
 		public static void UpdateMaxLength(this AppCompatEditText editText, int maxLength)
 		{
 			var currentFilters = new List<IInputFilter>(editText.GetFilters() ?? new IInputFilter[0]);
+			var changed = false;
 
 			for (var i = 0; i < currentFilters.Count; i++)
 			{
 				if (currentFilters[i] is InputFilterLengthFilter)
 				{
 					currentFilters.RemoveAt(i);
+					changed = true;
 					break;
 				}
 			}
 
-			currentFilters.Add(new InputFilterLengthFilter(maxLength));
+			if (maxLength > 0)
+			{
+				currentFilters.Add(new InputFilterLengthFilter(maxLength));
+				changed = true;
+			}
```

Comment:
A negative max length would just crash. And also make sure we only update when things have actually changed.